rllib/utils/serialization.py

```diff
@@ -325,3 +326,58 @@ def check_if_args_kwargs_serializable(args: Sequence[Any], kwargs: Dict[str, Any
                 f"Found non-serializable keyword argument: {k} = {v}.\n"
                 f"Original serialization error: {e}"
             )
+
+
+@DeveloperAPI
+def serialize_type(type_: Union[Type, str]) -> str:
+    """Converts a type into its full classpath ([module file] + "." + [class name]).
+    Args:
+        type_: The type to convert.
+    Returns:
+        The full classpath of the given type, e.g. "ray.rllib.algorithms.ppo.PPOConfig".
+    """
+    # Already serialized.
+    if isinstance(type_, str):
+        return type_
+
+    return type_.__module__ + "." + type_.__qualname__
+
+
+@DeveloperAPI
+def deserialize_type(
+    module: Union[str, Type], error: bool = False
+) -> Optional[Union[str, Type]]:
+    """Resolves a class path to a class.
+    If the given module is already a class, it is returned as is.
+    If the given module is a string, it is imported and the class is returned.
+    Args:
+        module: The classpath (str) or type to resolve.
+        error: Whether to throw a ValueError if `module` could not be resolved into
+            a class. If False and `module` is not resolvable, returns None.
+    Returns:
+        The resolved class or `module` (if `error` is False and no resolution possible).
+    Raises:
+        ValueError: If `error` is True and `module` cannot be resolved.
+    """
+    if isinstance(module, type):
+        return module
+
+    elif isinstance(module, str):
+        # Try interpreting (as classpath) and importing the given module.
+        try:
+            module_path, class_name = module.rsplit(".", 1)
+            module = importlib.import_module(module_path)
+            return getattr(module, class_name)
+        # Module not found.
+        except (ModuleNotFoundError, ImportError, AttributeError, ValueError) as e:
+            if error:
+                raise ValueError(
+                    f"Could not deserialize the given classpath `module={module}` into "
+                    "a valid python class! Make sure you have all necessary pip "
+                    "packages installed and all custom modules are in your "
+                    "`PYTHONPATH` env variable."
```

Comment:
I think we should be more verbose about Ray version/commit.
Since we are already saving that in the checkpoint (put it's not on the surface, not what user sees usually), we should raise other errors with the additional info that they might occur due to inconsistency between the used Ray versions. We might induce all sorts of errors here over time because we can't guarantee compatibility between versions. So then user would directly have a simple workaround (installing from specified commit) so they can get to run their checkpoint asap.

Reply:
I think the errors should instead get induced inside of `RLModule._from_metadata_file` when we move versions over time. 

That being said, we should probably maintain instead our own checkpoint versions internally, and a translation layer for them.